common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -191,13 +191,15 @@ def _publish_event(self, event_name, result, **kwargs):
             "result": result,
             "previous_count": getattr(self, "_last_event_result_count", len(self.selected)),
             "max_count": self.max_count,
+            "user_id": user_id,
+            "course_id": unicode(self.course_id),
```

Comment:
You're right. I had a memory of a conversation about it being okay to duplicate information that one would expect to be in the context of an event in the event payload if the information is critical to analyses that use the event, but now I can't find it. 

For what it's worth, I was working under the assumption that because this information is crucial to analysis that uses these events, it is preferable to have the information in the payload. Now that I look at the ticket again, though, I think that conversation occurred elsewhere and wasn't directly related to this ticket. Plus, without the course id in the event context, the events won't be include in the research package.

@mulby @brianhw do you have any thoughts on this?